ttnn/cpp/ttnn/tensor/xtensor/conversion_utils.cpp

```diff
@@ -0,0 +1,163 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "common/assert.hpp"
+#include "ttnn/cpp/ttnn/operations/copy.hpp"
+#include "ttnn/tensor/host_buffer/functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/xtensor/conversion_utils.hpp"
+#include <ttnn/tensor/xtensor/xtensor_all_includes.hpp>
+
+namespace ttnn::experimental::xtensor {
+namespace {
+
+using ::tt::tt_metal::Tensor;
+
+// copypaste from deprecated tensor pybinds ttnn
+tt::tt_metal::OwnedBuffer create_owned_buffer(const std::vector<float>& data, DataType data_type) {
```

Comment:
is data - a raw data or already partially prepared data? e.g. is it supposed to be tiled?

Reply:
It isn't raw data, and it is in row major layout. Does it make sense to allow providing buffer in non row major layout? Same goes for `to_vector` - I am assuming the primary use case is getting the data back in row major view?

Reply:
> Same goes for to_vector - I am assuming the primary use case is getting the data back in row major view?
Yes

> Does it make sense to allow providing buffer in non row major layout?
If you take data that was already prepared but is still on host. 
e.g. getting data from another tensor or getting some data that was serialized

Reply:
Ok. We can provide this as part of the `Tensor(Data data,  const TensorSpec&)` as per the comment below?